projects/legacy/components/input-date/input-date.component.ts

```diff
@@ -174,9 +174,13 @@ export class TuiInputDateComponent
         }
 
         this.value =
-            value.length !== DATE_FILLER_LENGTH
+            value.length !== DATE_FILLER_LENGTH || TUI_LETTER_REGEXP.test(value)
                 ? null
                 : TuiDay.normalizeParse(value, this.dateFormat.mode);
+
+        if (TUI_LETTER_REGEXP.test(this.nativeValue)) {
```

Comment:
do we really need an extra regex? looks like we can check if a string value is in the input using existing `this.activeItem` getter

```
 if (this.activeItem) {
      this.nativeValue = '';
 }

  this.value =
            value.length !== DATE_FILLER_LENGTH || this.activeItem
                ? null
                : TuiDay.normalizeParse(value, this.dateFormat.mode);
```

Reply:
Yes, it looks like it works in a similar way
Updated